# prepareChIPs

This is a simple `snakemake` workflow for preparing ChIP-Seq data.
The steps implemented are:

1. Download raw fastq files from SRA
2. Trim and Filter raw fastq files
3. Align to the supplied genome
4. Deduplicate Alignments
5. Call Macs2 Peaks

Full details for each step are given below.

## Download Raw Data

## Trim and Filter Reads

## Align Reads

## Deduplicate Reads

## Peak Calling