<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>QC: Aligned Data</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/core-js-2.5.3/shim.min.js"></script>
<script src="site_libs/react-17.0.0/react.min.js"></script>
<script src="site_libs/react-17.0.0/react-dom.min.js"></script>
<script src="site_libs/reactwidget-1.0.0/react-tools.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/reactable-0.4.4/reactable.css" rel="stylesheet" />
<script src="site_libs/reactable-binding-0.4.4/reactable.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">PRJNA509779</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    QC
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="raw_qc.html">Raw Data</a>
    </li>
    <li>
      <a href="trimmed_qc.html">Trimmed Data</a>
    </li>
    <li>
      <a href="align_qc.html">Alignments</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="AR_macs2_summary.html">AR</a>
    </li>
    <li>
      <a href="ERa_macs2_summary.html">ERa</a>
    </li>
    <li>
      <a href="H3K27ac_macs2_summary.html">H3K27ac</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/smped/prepareChIPs">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">QC: Aligned Data</h1>
<h4 class="date">05 June, 2023</h4>

</div>


<pre class="r"><code>library(tidyverse)
library(glue)
library(yaml)
library(here)
library(reactable)
library(pander)
library(ngsReports)
library(scales)
library(htmltools)
myTheme &lt;- theme(
    plot.title = element_text(hjust = 0.5),
    text = element_text(size = 13)
  )
## This manages a bug which exists in Bioc 3.16 but is fixed in ngsReports 2.2.3
.parseLogs &lt;- function(data, which = 1){
  {
    if (is.character(which)) 
        which &lt;- match.arg(which, c(&quot;metrics&quot;, &quot;histogram&quot;, &quot;all&quot;))
    if (is.numeric(which)) 
        stopifnot(which %in% c(1, 2))
    metrics &lt;- lapply(data, function(x) {
        cmd &lt;- stringr::str_subset(x, &quot;(INPUT|input)&quot;)
        ip &lt;- stringr::str_replace_all(cmd, &quot;.+(input|INPUT)([ =]*)([^ ]+.bam).+&quot;, 
            &quot;\\3&quot;)
        ip &lt;- str_replace_all(ip, &quot;[\\[\\]]&quot;, &quot;&quot;)
        metHeader &lt;- grep(&quot;METRICS CLASS\t.+.DuplicationMetrics&quot;, 
            x)
        df &lt;- ngsReports:::.splitByTab(x[seq(metHeader + 1, by = 1, length.out = 2)])
        df$LIBRARY &lt;- ip
        df
    })
    metrics &lt;- dplyr::bind_rows(metrics)
    metrics$PERCENT_DUPLICATION &lt;- as.numeric(metrics$PERCENT_DUPLICATION)
    intCols &lt;- setdiff(colnames(metrics), c(&quot;LIBRARY&quot;, &quot;PERCENT_DUPLICATION&quot;))
    metrics[intCols] &lt;- lapply(metrics[intCols], as.integer)
    metrics &lt;- as_tibble(metrics)
    histData &lt;- lapply(data, function(x) {
        cmd &lt;- stringr::str_subset(x, &quot;(INPUT|input)&quot;)
        ip &lt;- stringr::str_replace_all(cmd, &quot;.+(input|INPUT)([ =]*)([^ ]+.bam).+&quot;, 
            &quot;\\3&quot;)
        ip &lt;- str_replace_all(ip, &quot;[\\[\\]]&quot;, &quot;&quot;)
        histHeader &lt;- grep(&quot;HISTOGRAM\tjava.lang.Double&quot;, x)
        df &lt;- NULL
        if (length(histHeader)) {
            x &lt;- x[-seq_len(histHeader)]
            x &lt;- x[!grepl(&quot;^$&quot;, x)]
            df &lt;- ngsReports:::.splitByTab(x)
            df$LIBRARY &lt;- ip
            df &lt;- dplyr::select(df, &quot;LIBRARY&quot;, everything())
        }
        df
    })
    histData &lt;- dplyr::bind_rows(histData)
    if (ncol(histData)) {
        histData &lt;- lapply(histData, function(x) {
            any_na &lt;- any(is.na(suppressWarnings(as.numeric(x))))
            if (!any_na) 
                x &lt;- as.numeric(x)
            x
        })
        histData$BIN &lt;- as.integer(histData$BIN)
        histData &lt;- as_tibble(histData)
    }
    out &lt;- list(metrics = metrics, histogram = histData)
    out &lt;- lapply(out, function(x) {
        colnames(x) &lt;- stringr::str_replace_all(colnames(x), 
            &quot;_&quot;, &quot; &quot;)
        colnames(x) &lt;- stringr::str_to_title(colnames(x))
        x
    })
    if (which != &quot;all&quot;) 
        out &lt;- out[[which]]
    out
  }
}</code></pre>
<pre class="r"><code>config &lt;- read_yaml(here::here(&quot;config/config.yml&quot;))
samples &lt;- read_tsv(here::here(config$samples)) %&gt;% 
  bind_rows(
    tibble(
      accession = unique(.$input), target = &quot;Input&quot;
    )
  )
bwt2_logs &lt;- here::here(&quot;output&quot;, &quot;bowtie2&quot;, glue(&quot;{samples$accession}.log&quot;)) %&gt;% 
  setNames(samples$accession) %&gt;% 
  .[file.exists(.)] %&gt;% 
  importNgsLogs() 
duplogs &lt;- here::here(&quot;output&quot;, &quot;markDuplicates&quot;, glue(&quot;{samples$accession}.metrics.txt&quot;)) %&gt;%
  setNames(samples$accession) %&gt;% 
  .[file.exists(.)] %&gt;% 
  sapply(readLines, simplify = FALSE) %&gt;% 
  .parseLogs() %&gt;% 
  mutate(
    Filename = gsub(&quot;.sorted.+&quot;, &quot;&quot;, basename(Library))
  ) %&gt;% 
  dplyr::select(Filename, everything())
qc_path &lt;- here::here(config$paths$qc, &quot;raw&quot;)
fl &lt;- file.path(qc_path, glue(&quot;{samples$accession}_fastqc.zip&quot;)) %&gt;% 
  setNames(samples$accession)
rawFqc &lt;- FastqcDataList(fl[file.exists(fl)])
names(rawFqc) &lt;- names(fl)[file.exists(fl)]
samples &lt;- readTotals(rawFqc) %&gt;% 
  mutate(accession = str_remove_all(Filename, &quot;.fastq.gz&quot;)) %&gt;% 
  dplyr::select(accession, Total_Sequences) %&gt;% 
  right_join(samples, by = &quot;accession&quot;)</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This final QC worksheet summarises the alignments and de-duplication
metrics. Single-end alignments were performed using <code>bowtie2</code>
<span class="citation">(<a href="#ref-Langmead2012-qc"
role="doc-biblioref">Langmead and Salzberg 2012</a>)</span> before being
de-duplicated using <code>markDuplicates</code> from the Picard suite of
tools (<a href="https://broadinstitute.github.io/picard"
class="uri">https://broadinstitute.github.io/picard</a>)</p>
<div id="alignment-rates" class="section level3">
<h3>Alignment Rates</h3>
<pre class="r"><code>bwt2_logs %&gt;% 
  mutate(Filename = str_remove_all(Filename, &quot;.log&quot;)) %&gt;% 
  dplyr::select(Filename, Unique_Unpaired, Multiple_Unpaired, Not_Aligned) %&gt;% 
  pivot_longer(cols = contains(&quot;_&quot;), names_to = &quot;Status&quot;, values_to = &quot;Reads&quot;) %&gt;% 
  left_join(samples, by = c(&quot;Filename&quot; = &quot;accession&quot;)) %&gt;% 
  mutate(
    Status = Status %&gt;% 
      str_remove_all(&quot;_Unpaired&quot;) %&gt;% 
      str_replace_all(&quot;_&quot;, &quot; &quot;) %&gt;% 
      factor() %&gt;%
      relevel(ref = &quot;Unique&quot;),
    treatment = str_replace_na(treatment, &quot;&quot;)
  ) %&gt;% 
  arrange(Status) %&gt;% 
  mutate(cumsum = cumsum(Reads), p = Reads / sum(Reads), .by = &quot;Filename&quot;) %&gt;% 
  mutate(y = cumsum - 0.5 * Reads) %&gt;% 
  ggplot(aes(Filename, Reads, fill = fct_rev(Status))) +
  geom_col() +
  geom_label(
    aes(Filename, y, label = percent(p, 0.1)),
    data = . %&gt;% dplyr::filter(p &gt; 0.05),
    fill = &quot;white&quot;, alpha = 0.6, show.legend = FALSE
  ) +
  facet_grid(target + treatment ~ ., scales = &quot;free&quot;, space = &quot;free&quot;) +
  scale_y_continuous(
    labels = comma_format(scale = 1e-6), expand = expansion(c(0, 0.05))
  ) +
  scale_fill_viridis_d(direction = -1) +
  labs(y = &quot;Reads (millions)&quot;, fill = &quot;Alignment\nStatus&quot;) +
  theme_bw() +
  myTheme +
  coord_flip()</code></pre>
<p><img src="align_qc_files/figure-html/plot-aln-summary-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="deduplication-summary" class="section level2">
<h2>DeDuplication Summary</h2>
<pre class="r"><code>duplogs %&gt;% 
    dplyr::select(
      Filename, Alignments = `Unpaired Reads Examined` , 
      Duplicates = `Unpaired Read Duplicates`, `Percent Duplication`  
    ) %&gt;% 
    left_join(samples, by = c(&quot;Filename&quot; = &quot;accession&quot;)) %&gt;% 
    dplyr::select(-input) %&gt;% 
    mutate(
      Alignments = Alignments - Duplicates,
      Discarded = Total_Sequences - Alignments - Duplicates
    ) %&gt;% 
    dplyr::select(
      Filename, Total_Sequences, Alignments, Discarded, Duplicates, everything()
    ) %&gt;% 
  pivot_longer(
    cols = all_of(c(&quot;Alignments&quot;, &quot;Discarded&quot;, &quot;Duplicates&quot;)),
    names_to = &quot;Status&quot;, values_to = &quot;Reads&quot;
  ) %&gt;% 
  mutate(
    Status = factor(Status, levels = c(&quot;Discarded&quot;, &quot;Duplicates&quot;, &quot;Alignments&quot;)),
    treatment = str_replace_na(treatment, &quot;&quot;)
  ) %&gt;% 
  arrange(Filename, desc(Status)) %&gt;% 
  mutate(
    p = Reads / sum(Reads),
    y = cumsum(Reads) - 0.5 * Reads,
    .by = Filename
  ) %&gt;% 
  ggplot(aes(Filename, Reads, fill = Status )) +
  geom_col() +
  geom_label(
    aes(Filename, y, label = percent(p, 0.1)),
    data = . %&gt;% dplyr::filter(p &gt; 0.033),
    fill = &quot;white&quot;, alpha = 0.7, size = 3.5
  ) +
  facet_grid(target + treatment ~ ., scales = &quot;free&quot;, space = &quot;free&quot;) +
  scale_y_continuous(
    labels = comma_format(scale = 1e-6), expand = expansion(c(0, 0.05))
  ) +
  scale_fill_viridis_d(direction = -1) +
  labs(y = &quot;Reads (millions)&quot;, fill = &quot;Alignment\nStatus&quot;) +
  theme_bw() +
  myTheme +
  coord_flip()</code></pre>
<div class="figure" style="text-align: center">
<img src="align_qc_files/figure-html/dedup-summary-1.png" alt="*Summary of the final alignments after trimming, filtering and deduplicating. Percentages are shown only if greater than 3.3% to avoid overlapping labels*" width="960" />
<p class="caption">
<em>Summary of the final alignments after trimming, filtering and
deduplicating. Percentages are shown only if greater than 3.3% to avoid
overlapping labels</em>
</p>
</div>
</div>
<div id="summary-table" class="section level2">
<h2>Summary Table</h2>
<pre class="r"><code>div(
  class = &quot;table&quot;,
  div(
    class = &quot;table-header&quot;,
    htmltools::tags$caption(
      htmltools::em(
        &quot;Alignment and de-duplication summary, with the number of reads in the 
        initial fastq files compared to the total number of retained alignments 
        after de-duplication.&quot;
      )
    )
  ),
  duplogs %&gt;% 
    dplyr::select(
      Filename, Alignments = `Unpaired Reads Examined` , 
      Duplicates = `Unpaired Read Duplicates`, `Percent Duplication`  
    ) %&gt;% 
    left_join(samples, by = c(&quot;Filename&quot; = &quot;accession&quot;)) %&gt;% 
    dplyr::select(-input) %&gt;% 
    mutate(
      Alignments = Alignments - Duplicates,
      `% Retained` = Alignments / Total_Sequences,
      Discarded = Total_Sequences - Alignments - Duplicates
    ) %&gt;% 
    dplyr::select(
      Filename, Total_Sequences, Alignments, Discarded, Duplicates, `% Retained`, 
      everything()
    ) %&gt;% 
    setNames(str_replace_all(names(.), &quot;_&quot;, &quot; &quot;)) %&gt;% 
    setNames(str_to_title(names(.))) %&gt;% 
    reactable(
      sortable = TRUE, filterable = TRUE, resizable = TRUE,
      showPageSizeOptions = TRUE,
      columns = list(
        `Total Sequences` = colDef(
          name = &quot;Raw Reads&quot;, format = colFormat(separators = TRUE)
        ),
        Discarded = colDef(format = colFormat(separators = TRUE)),
        Alignments = colDef(format = colFormat(separators = TRUE)),
        `% Retained` = colDef(format = colFormat(percent = TRUE, digits = 2)),
        Duplicates = colDef(format = colFormat(separators = TRUE)),
        `Percent Duplication` = colDef(
          name = &quot;% Duplication&quot;,
          format = colFormat(percent = TRUE, digits = 2)
        )
      )
    )
)</code></pre>
<div class="table">
<div class="table-header">
<caption>
<em>Alignment and de-duplication summary, with the number of reads in the 
        initial fastq files compared to the total number of retained alignments 
        after de-duplication.</em>
</caption>
</div>
<div class="reactable html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-2a85c69757f1c9eab313" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-2a85c69757f1c9eab313">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"Filename":["SRR8315174","SRR8315175","SRR8315176","SRR8315177","SRR8315178","SRR8315179","SRR8315180","SRR8315181","SRR8315182","SRR8315183","SRR8315184","SRR8315185","SRR8315186","SRR8315187","SRR8315188","SRR8315189","SRR8315190","SRR8315191","SRR8315192"],"Total Sequences":[20304990,23312643,16701719,20618308,21273354,21207946,19871195,21393665,21524645,19702604,22258868,21989625,18193552,21347628,20268206,20356530,22174536,19493685,23022741],"Alignments":[16929689,19412922,13930124,17173150,17715794,17729490,16373508,17581871,17641037,16208256,18270619,18061306,14881059,17185932,16246941,16401078,17955212,15842936,19309861],"Discarded":[2704736,3075954,2225812,2685226,2780802,2767204,2673759,2920308,2883717,2701200,2986472,3004193,1987380,2447252,2398668,2371213,2407507,2142357,2928012],"Duplicates":[670565,823767,545783,759932,776758,711252,823928,891486,999891,793148,1001777,924126,1325113,1714444,1622597,1584239,1811817,1508392,784868],"% Retained":[0.833769876271793,0.832720768726223,0.83405330912345,0.832907821534143,0.832769200380908,0.835983362085135,0.823982050400089,0.821826040559203,0.819573888442759,0.822645372154868,0.820824266535028,0.821355798473144,0.817930385446448,0.805051127928592,0.801597388540456,0.80569124502064,0.80972210647384,0.812721453127,0.83872988885207],"Percent Duplication":[0.0381,0.040707,0.037703,0.042376,0.042004,0.03857,0.04791,0.048258,0.05364,0.046652,0.05198,0.048676,0.081766,0.09071,0.090802,0.088085,0.091659,0.086932,0.039058],"Target":["AR","AR","AR","AR","AR","AR","ERa","ERa","ERa","ERa","ERa","ERa","H3K27ac","H3K27ac","H3K27ac","H3K27ac","H3K27ac","H3K27ac","Input"],"Treatment":["E2","E2","E2","E2DHT","E2DHT","E2DHT","E2","E2","E2","E2DHT","E2DHT","E2DHT","E2","E2","E2","E2DHT","E2DHT","E2DHT",null],"Cell Line":["ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1","ZR-75-1",null],"Sample Name":["GSM3511082","GSM3511082","GSM3511082","GSM3511083","GSM3511083","GSM3511083","GSM3511084","GSM3511084","GSM3511084","GSM3511085","GSM3511085","GSM3511085","GSM3511086","GSM3511086","GSM3511086","GSM3511087","GSM3511087","GSM3511087",null],"Experiment":["SRX5128130","SRX5128130","SRX5128130","SRX5128131","SRX5128131","SRX5128131","SRX5128132","SRX5128132","SRX5128132","SRX5128133","SRX5128133","SRX5128133","SRX5128134","SRX5128134","SRX5128134","SRX5128135","SRX5128135","SRX5128135",null],"Source Name":["ZR-75-1_E2","ZR-75-1_E2","ZR-75-1_E2","ZR-75-1_E2+DHT","ZR-75-1_E2+DHT","ZR-75-1_E2+DHT","ZR-75-1_E2","ZR-75-1_E2","ZR-75-1_E2","ZR-75-1_E2+DHT","ZR-75-1_E2+DHT","ZR-75-1_E2+DHT","ZR-75-1_E2","ZR-75-1_E2","ZR-75-1_E2","ZR-75-1_E2+DHT","ZR-75-1_E2+DHT","ZR-75-1_E2+DHT",null]},"columns":[{"id":"Filename","name":"Filename","type":"character"},{"id":"Total Sequences","name":"Raw Reads","type":"numeric","format":{"cell":{"separators":true},"aggregated":{"separators":true}}},{"id":"Alignments","name":"Alignments","type":"numeric","format":{"cell":{"separators":true},"aggregated":{"separators":true}}},{"id":"Discarded","name":"Discarded","type":"numeric","format":{"cell":{"separators":true},"aggregated":{"separators":true}}},{"id":"Duplicates","name":"Duplicates","type":"numeric","format":{"cell":{"separators":true},"aggregated":{"separators":true}}},{"id":"% Retained","name":"% Retained","type":"numeric","format":{"cell":{"digits":2,"percent":true},"aggregated":{"digits":2,"percent":true}}},{"id":"Percent Duplication","name":"% Duplication","type":"numeric","format":{"cell":{"digits":2,"percent":true},"aggregated":{"digits":2,"percent":true}}},{"id":"Target","name":"Target","type":"character"},{"id":"Treatment","name":"Treatment","type":"character"},{"id":"Cell Line","name":"Cell Line","type":"character"},{"id":"Sample Name","name":"Sample Name","type":"character"},{"id":"Experiment","name":"Experiment","type":"character"},{"id":"Source Name","name":"Source Name","type":"character"}],"resizable":true,"filterable":true,"showPageSizeOptions":true,"dataKey":"37f013a9fe4657edf3fedd78ef48b09b"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Langmead2012-qc" class="csl-entry">
Langmead, Ben, and Steven L Salzberg. 2012. <span>“Fast Gapped-Read
Alignment with Bowtie 2.”</span> <em>Nat. Methods</em> 9 (4): 357–59.
</div>
</div>
<br>
<button type="button" class="btn btn-default btn-sessioninfo" data-toggle="collapse" data-target="#sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-info-sign" aria-hidden="true"></span>
Session information </button>
</p>
<div id="sessioninfo" class="collapse">
<p><strong>R version 4.3.0 (2023-04-21)</strong></p>
<p><strong>Platform:</strong> x86_64-pc-linux-gnu (64-bit)</p>
<p><strong>locale:</strong> <em>LC_CTYPE=en_AU.UTF-8</em>,
<em>LC_NUMERIC=C</em>, <em>LC_TIME=en_AU.UTF-8</em>,
<em>LC_COLLATE=en_AU.UTF-8</em>, <em>LC_MONETARY=en_AU.UTF-8</em>,
<em>LC_MESSAGES=en_AU.UTF-8</em>, <em>LC_PAPER=en_AU.UTF-8</em>,
<em>LC_NAME=C</em>, <em>LC_ADDRESS=C</em>, <em>LC_TELEPHONE=C</em>,
<em>LC_MEASUREMENT=en_AU.UTF-8</em> and <em>LC_IDENTIFICATION=C</em></p>
<p><strong>attached base packages:</strong> <em>stats</em>,
<em>graphics</em>, <em>grDevices</em>, <em>utils</em>,
<em>datasets</em>, <em>methods</em> and <em>base</em></p>
<p><strong>other attached packages:</strong> <em>magrittr(v.2.0.3)</em>,
<em>DiagrammeR(v.1.0.10)</em>, <em>Polychrome(v.1.5.1)</em>,
<em>htmltools(v.0.5.5)</em>, <em>scales(v.1.2.1)</em>,
<em>ngsReports(v.2.3.3)</em>, <em>patchwork(v.1.1.2)</em>,
<em>BiocGenerics(v.0.46.0)</em>, <em>pander(v.0.6.5)</em>,
<em>reactable(v.0.4.4)</em>, <em>here(v.1.0.1)</em>,
<em>yaml(v.2.3.7)</em>, <em>glue(v.1.6.2)</em>,
<em>lubridate(v.1.9.2)</em>, <em>forcats(v.1.0.0)</em>,
<em>stringr(v.1.5.0)</em>, <em>dplyr(v.1.1.2)</em>,
<em>purrr(v.1.0.1)</em>, <em>readr(v.2.1.4)</em>,
<em>tidyr(v.1.3.0)</em>, <em>tibble(v.3.2.1)</em>,
<em>ggplot2(v.3.4.2)</em> and <em>tidyverse(v.2.0.0)</em></p>
<p><strong>loaded via a namespace (and not attached):</strong>
<em>tidyselect(v.1.2.0)</em>, <em>viridisLite(v.0.4.2)</em>,
<em>farver(v.2.1.1)</em>, <em>Biostrings(v.2.68.1)</em>,
<em>bitops(v.1.0-7)</em>, <em>fastmap(v.1.1.1)</em>,
<em>RCurl(v.1.98-1.12)</em>, <em>lazyeval(v.0.2.2)</em>,
<em>digest(v.0.6.31)</em>, <em>timechange(v.0.2.0)</em>,
<em>lifecycle(v.1.0.3)</em>, <em>ellipsis(v.0.3.2)</em>,
<em>compiler(v.4.3.0)</em>, <em>rlang(v.1.1.1)</em>,
<em>sass(v.0.4.6)</em>, <em>tools(v.4.3.0)</em>, <em>utf8(v.1.2.3)</em>,
<em>data.table(v.1.14.8)</em>, <em>knitr(v.1.43)</em>,
<em>labeling(v.0.4.2)</em>, <em>htmlwidgets(v.1.6.2)</em>,
<em>bit(v.4.0.5)</em>, <em>scatterplot3d(v.0.3-44)</em>,
<em>RColorBrewer(v.1.1-3)</em>, <em>plyr(v.1.8.8)</em>,
<em>withr(v.2.5.0)</em>, <em>grid(v.4.3.0)</em>,
<em>stats4(v.4.3.0)</em>, <em>fansi(v.1.0.4)</em>,
<em>colorspace(v.2.1-0)</em>, <em>MASS(v.7.3-60)</em>,
<em>cli(v.3.6.1)</em>, <em>rmarkdown(v.2.22)</em>,
<em>crayon(v.1.5.2)</em>, <em>generics(v.0.1.3)</em>,
<em>rstudioapi(v.0.14)</em>, <em>httr(v.1.4.6)</em>,
<em>reshape2(v.1.4.4)</em>, <em>tzdb(v.0.4.0)</em>,
<em>visNetwork(v.2.1.2)</em>, <em>cachem(v.1.0.8)</em>,
<em>zlibbioc(v.1.46.0)</em>, <em>parallel(v.4.3.0)</em>,
<em>XVector(v.0.40.0)</em>, <em>vctrs(v.0.6.2)</em>,
<em>jsonlite(v.1.8.4)</em>, <em>IRanges(v.2.34.0)</em>,
<em>hms(v.1.1.3)</em>, <em>S4Vectors(v.0.38.1)</em>,
<em>bit64(v.4.0.5)</em>, <em>crosstalk(v.1.2.0)</em>,
<em>fontawesome(v.0.5.1)</em>, <em>plotly(v.4.10.1)</em>,
<em>jquerylib(v.0.1.4)</em>, <em>ggdendro(v.0.1.23)</em>,
<em>reactR(v.0.4.4)</em>, <em>DT(v.0.28)</em>,
<em>stringi(v.1.7.12)</em>, <em>gtable(v.0.3.3)</em>,
<em>GenomeInfoDb(v.1.36.0)</em>, <em>munsell(v.0.5.0)</em>,
<em>pillar(v.1.9.0)</em>, <em>GenomeInfoDbData(v.1.2.10)</em>,
<em>R6(v.2.5.1)</em>, <em>rprojroot(v.2.0.3)</em>,
<em>vroom(v.1.6.3)</em>, <em>evaluate(v.0.21)</em>,
<em>lattice(v.0.21-8)</em>, <em>highr(v.0.10)</em>,
<em>bslib(v.0.4.2)</em>, <em>Rcpp(v.1.0.10)</em>, <em>xfun(v.0.39)</em>,
<em>zoo(v.1.8-12)</em> and <em>pkgconfig(v.2.0.3)</em></p>
</div>
</div>

<div class="footer">
  <br>Workflow developed by Dr Stevie Pederson<sup>1,2,3</sup><br><br>
  <sup>1.</sup>Black Ochre Data Laboratories, Telethon Kids Institute, Adelaide, SA, Australia<br>
  <sup>2.</sup>John Curtin School of Medical Resarch, Australian National University, Canberra, ACT, Australia<br>
  <sup>3.</sup>Dame Roma Mitchell Cancer Research Laboratories, Adelaide Medical School, University of Adelaide, SA, Australia<br>
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
