---
title: "QC: Aligned Data"
date: "`r format(Sys.Date(), '%d %B, %Y')`"
bibliography: references.bib
link-citations: true
---

```{r setup, echo = FALSE}
conda_pre <- system2("echo", "$CONDA_PREFIX", stdout = TRUE)
if (conda_pre != "") {
  conda_lib_path <- file.path(conda_pre, "lib", "R", "library")
  if (!dir.exists(conda_lib_path)) conda_lib_path <- NULL
  prev_paths <- .libPaths()
  paths_to_set <- unique(c(conda_lib_path, prev_paths))
  .libPaths(paths_to_set)
}
knitr::opts_chunk$set(
  message = FALSE, warning = FALSE,
  fig.height = 8, fig.width = 10, fig.align = 'center'
)
```

```{r packages}
library(tidyverse)
library(glue)
library(yaml)
library(here)
library(reactable)
library(pander)
library(ngsReports)
library(scales)
library(htmltools)
myTheme <- theme(
    plot.title = element_text(hjust = 0.5),
    text = element_text(size = 13)
  )
## This manages a bug which exists in Bioc 3.16 but is fixed in ngsReports 2.2.3
.parseLogs <- function(data, which = 1){
  {
    if (is.character(which)) 
        which <- match.arg(which, c("metrics", "histogram", "all"))
    if (is.numeric(which)) 
        stopifnot(which %in% c(1, 2))
    metrics <- lapply(data, function(x) {
        cmd <- stringr::str_subset(x, "(INPUT|input)")
        ip <- stringr::str_replace_all(cmd, ".+(input|INPUT)([ =]*)([^ ]+.bam).+", 
            "\\3")
        ip <- str_replace_all(ip, "[\\[\\]]", "")
        metHeader <- grep("METRICS CLASS\t.+.DuplicationMetrics", 
            x)
        df <- ngsReports:::.splitByTab(x[seq(metHeader + 1, by = 1, length.out = 2)])
        df$LIBRARY <- ip
        df
    })
    metrics <- dplyr::bind_rows(metrics)
    metrics$PERCENT_DUPLICATION <- as.numeric(metrics$PERCENT_DUPLICATION)
    intCols <- setdiff(colnames(metrics), c("LIBRARY", "PERCENT_DUPLICATION"))
    metrics[intCols] <- lapply(metrics[intCols], as.integer)
    metrics <- as_tibble(metrics)
    histData <- lapply(data, function(x) {
        cmd <- stringr::str_subset(x, "(INPUT|input)")
        ip <- stringr::str_replace_all(cmd, ".+(input|INPUT)([ =]*)([^ ]+.bam).+", 
            "\\3")
        ip <- str_replace_all(ip, "[\\[\\]]", "")
        histHeader <- grep("HISTOGRAM\tjava.lang.Double", x)
        df <- NULL
        if (length(histHeader)) {
            x <- x[-seq_len(histHeader)]
            x <- x[!grepl("^$", x)]
            df <- ngsReports:::.splitByTab(x)
            df$LIBRARY <- ip
            df <- dplyr::select(df, "LIBRARY", everything())
        }
        df
    })
    histData <- dplyr::bind_rows(histData)
    if (ncol(histData)) {
        histData <- lapply(histData, function(x) {
            any_na <- any(is.na(suppressWarnings(as.numeric(x))))
            if (!any_na) 
                x <- as.numeric(x)
            x
        })
        histData$BIN <- as.integer(histData$BIN)
        histData <- as_tibble(histData)
    }
    out <- list(metrics = metrics, histogram = histData)
    out <- lapply(out, function(x) {
        colnames(x) <- stringr::str_replace_all(colnames(x), 
            "_", " ")
        colnames(x) <- stringr::str_to_title(colnames(x))
        x
    })
    if (which != "all") 
        out <- out[[which]]
    out
  }
}
```

```{r load-data}
config <- read_yaml(here::here("config/config.yml"))
samples <- read_tsv(here::here(config$samples)) %>% 
  bind_rows(
    tibble(
      accession = unique(.$input), target = "Input"
    )
  )
bwt2_logs <- here::here("output", "bowtie2", glue("{samples$accession}.log")) %>% 
  setNames(samples$accession) %>% 
  .[file.exists(.)] %>% 
  importNgsLogs() 
duplogs <- here::here("output", "markDuplicates", glue("{samples$accession}.metrics.txt")) %>%
  setNames(samples$accession) %>% 
  .[file.exists(.)] %>% 
  sapply(readLines, simplify = FALSE) %>% 
  .parseLogs() %>% 
  mutate(
    Filename = gsub(".sorted.+", "", basename(Library))
  ) %>% 
  dplyr::select(Filename, everything())
qc_path <- here::here(config$paths$qc, "raw")
fl <- file.path(qc_path, glue("{samples$accession}_fastqc.zip")) %>% 
  setNames(samples$accession)
rawFqc <- FastqcDataList(fl[file.exists(fl)])
names(rawFqc) <- names(fl)[file.exists(fl)]
samples <- readTotals(rawFqc) %>% 
  mutate(accession = str_remove_all(Filename, ".fastq.gz")) %>% 
  dplyr::select(accession, Total_Sequences) %>% 
  right_join(samples, by = "accession")
```

## Introduction

This final QC worksheet summarises the alignments and de-duplication metrics.
Single-end alignments were performed using `bowtie2` [@Langmead2012-qc] before being de-duplicated using `markDuplicates` from the Picard suite of tools (https://broadinstitute.github.io/picard)


### Alignment Rates

```{r plot-aln-summary, eval = length(bwt2_logs) > 0}
bwt2_logs %>% 
  mutate(Filename = str_remove_all(Filename, ".log")) %>% 
  dplyr::select(Filename, Unique_Unpaired, Multiple_Unpaired, Not_Aligned) %>% 
  pivot_longer(cols = contains("_"), names_to = "Status", values_to = "Reads") %>% 
  left_join(samples, by = c("Filename" = "accession")) %>% 
  mutate(
    Status = Status %>% 
      str_remove_all("_Unpaired") %>% 
      str_replace_all("_", " ") %>% 
      factor() %>%
      relevel(ref = "Unique"),
    treatment = str_replace_na(treatment, "")
  ) %>% 
  arrange(Status) %>% 
  mutate(cumsum = cumsum(Reads), p = Reads / sum(Reads), .by = "Filename") %>% 
  mutate(y = cumsum - 0.5 * Reads) %>% 
  ggplot(aes(Filename, Reads, fill = fct_rev(Status))) +
  geom_col() +
  geom_label(
    aes(Filename, y, label = percent(p, 0.1)),
    data = . %>% dplyr::filter(p > 0.05),
    fill = "white", alpha = 0.6, show.legend = FALSE
  ) +
  facet_grid(target + treatment ~ ., scales = "free", space = "free") +
  scale_y_continuous(
    labels = comma_format(scale = 1e-6), expand = expansion(c(0, 0.05))
  ) +
  scale_fill_viridis_d(direction = -1) +
  labs(y = "Reads (millions)", fill = "Alignment\nStatus") +
  theme_bw() +
  myTheme +
  coord_flip()
```

## DeDuplication Summary

```{r dedup-summary, fig.cap = "*Summary of the final alignments after trimming, filtering and deduplicating. Percentages are shown only if greater than 3.3% to avoid overlapping labels*"}
duplogs %>% 
    dplyr::select(
      Filename, Alignments = `Unpaired Reads Examined` , 
      Duplicates = `Unpaired Read Duplicates`, `Percent Duplication`  
    ) %>% 
    left_join(samples, by = c("Filename" = "accession")) %>% 
    dplyr::select(-input) %>% 
    mutate(
      Alignments = Alignments - Duplicates,
      Discarded = Total_Sequences - Alignments - Duplicates
    ) %>% 
    dplyr::select(
      Filename, Total_Sequences, Alignments, Discarded, Duplicates, everything()
    ) %>% 
  pivot_longer(
    cols = all_of(c("Alignments", "Discarded", "Duplicates")),
    names_to = "Status", values_to = "Reads"
  ) %>% 
  mutate(
    Status = factor(Status, levels = c("Discarded", "Duplicates", "Alignments")),
    treatment = str_replace_na(treatment, "")
  ) %>% 
  arrange(Filename, desc(Status)) %>% 
  mutate(
    p = Reads / sum(Reads),
    y = cumsum(Reads) - 0.5 * Reads,
    .by = Filename
  ) %>% 
  ggplot(aes(Filename, Reads, fill = Status )) +
  geom_col() +
  geom_label(
    aes(Filename, y, label = percent(p, 0.1)),
    data = . %>% dplyr::filter(p > 0.033),
    fill = "white", alpha = 0.7, size = 3.5
  ) +
  facet_grid(target + treatment ~ ., scales = "free", space = "free") +
  scale_y_continuous(
    labels = comma_format(scale = 1e-6), expand = expansion(c(0, 0.05))
  ) +
  scale_fill_viridis_d(direction = -1) +
  labs(y = "Reads (millions)", fill = "Alignment\nStatus") +
  theme_bw() +
  myTheme +
  coord_flip()
```

## Summary Table

```{r tbl-duplication}
div(
  class = "table",
  div(
    class = "table-header",
    htmltools::tags$caption(
      htmltools::em(
        "Alignment and de-duplication summary, with the number of reads in the 
        initial fastq files compared to the total number of retained alignments 
        after de-duplication."
      )
    )
  ),
  duplogs %>% 
    dplyr::select(
      Filename, Alignments = `Unpaired Reads Examined` , 
      Duplicates = `Unpaired Read Duplicates`, `Percent Duplication`  
    ) %>% 
    left_join(samples, by = c("Filename" = "accession")) %>% 
    dplyr::select(-input) %>% 
    mutate(
      Alignments = Alignments - Duplicates,
      `% Retained` = Alignments / Total_Sequences,
      Discarded = Total_Sequences - Alignments - Duplicates
    ) %>% 
    dplyr::select(
      Filename, Total_Sequences, Alignments, Discarded, Duplicates, `% Retained`, 
      everything()
    ) %>% 
    setNames(str_replace_all(names(.), "_", " ")) %>% 
    setNames(str_to_title(names(.))) %>% 
    reactable(
      sortable = TRUE, filterable = TRUE, resizable = TRUE,
      showPageSizeOptions = TRUE,
      columns = list(
        `Total Sequences` = colDef(
          name = "Raw Reads", format = colFormat(separators = TRUE)
        ),
        Discarded = colDef(format = colFormat(separators = TRUE)),
        Alignments = colDef(format = colFormat(separators = TRUE)),
        `% Retained` = colDef(format = colFormat(percent = TRUE, digits = 2)),
        Duplicates = colDef(format = colFormat(separators = TRUE)),
        `Percent Duplication` = colDef(
          name = "% Duplication",
          format = colFormat(percent = TRUE, digits = 2)
        )
      )
    )
)
```

## References

<div id="refs"></div>

<br>
<button type="button" class="btn btn-default btn-sessioninfo" data-toggle="collapse" data-target="#sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-info-sign" aria-hidden="true"></span> Session information
</button>
</p>
<div id="sessioninfo" class="collapse">
```{r session-info, echo=FALSE}
pander::pander(sessionInfo())
```
</div>
